### ISSUE TITLE
The conditional check 'persistent_volumes | length > 0 or persistent_volume_claims | length > 0' failed.

### ISSUE THREAD

**Issue by magick93** [openshift/openshift-ansible#3855]
#### Description

Am getting the following error when trying to install OpenShift using openshift-ansible.
I tried the workaround mentioned on https://github.com/openshift/openshift-ansible/issues/2553#issuecomment-256351225 but it did not work.

```
TASK [openshift_sanitize_inventory : Standardize on latest variable names] *****
fatal: [console.mydomain.com]: FAILED! => {"failed": true, "msg": "The conditional check 'persistent_volumes | length > 0 or persistent_volume_claims | length > 0' failed. The error was: {{ hostvars[groups.oo_first_master.0] | oo_persistent_volumes(groups) }}: |failed no storage host detected"}
	to retry, use: --limit @/root/openshift-ansible/playbooks/byo/config.retry
```

##### Version

**openshift-ansible Version**
openshift-ansible-3.6.16-1

**Ansible Version**
ansible 2.2.0.0

**Config**

```
[OSEv3:children]
masters
nodes

[OSEv3:vars]
ansible_ssh_user=root
deployment_type=origin
openshift_master_identity_providers=[{'name': 'htpasswd_auth', 'login': 'true', 'challenge': 'true', 'kind': 'HTPasswdPasswordIdentityProvider', 'filename': '/etc/origin/master/htpasswd'}]
containerized=true
openshift_release=v1.5.0
openshift_image_tag=v1.5.0-rc.0
openshift_public_hostname=console.mydomain.com
openshift_master_default_subdomain=apps.console.mydomain.com
openshift_hosted_metrics_deploy=false #originally set this to true, but got same error
enable_excluders=false 


#openshift_hosted_logging_deployer_version=v3.4 #also tried 3.4.0
#openshift_hosted_logging_deployer_version=v3.4


openshift_hosted_metrics_storage_kind=nfs
openshift_hosted_metrics_storage_access_modes=['ReadWriteOnce']
openshift_hosted_metrics_storage_nfs_directory=/exports
openshift_hosted_metrics_storage_nfs_options='*(rw,root_squash)'
openshift_hosted_metrics_storage_volume_name=metrics
openshift_hosted_metrics_storage_volume_size=10Gi

[masters]
console.mydomain.com openshift_schedulable=true

[nodes]
console.mydomain.com openshift_node_labels="{'region': 'infra', 'zone': 'default'}" openshift_schedulable=true

```

##### Expected Results
Expect that this error doesnt occur. 

##### Additional Information

** Operating system and version**
CentOS Linux release 7.3.1611 (Core)

**Comment by sdodson** [comment]
@abutcher didn't you figure this out in the past?

**Comment by magick93** [comment]
Am still getting this error on `openshift-ansible-3.6.17-1-12-g3584659`.

**Comment by abutcher** [comment]
Hey @magick93, no `[nfs]` group or `openshift_hosted_metrics_storage_host=nfs.example.com` has been defined in the inventory so this is an expected error message. You should be able to specify an `nfs` group with a single host or point to an external nfs server where an `/exports/metrics` volume exists.

**Comment by debianmaster** [comment]
seeing same issue, any workarounds?

**Comment by eformat** [comment]
this worked for me - set logging to be dynamic for starters:

openshift_hosted_logging_storage_kind=dynamic

**Comment by mmckinst** [comment]
I was also getting this error message on OCP 3.7. The problem is related to openshift_facts.py as described in https://github.com/openshift/openshift-ansible/issues/6263 . The problem was fixed in https://github.com/openshift/openshift-ansible/pull/6272 .

If you run your installer with `-vvv` and look at the logs, look for `ansible_facts` and inside that you'll find a section for metrics where the nfs host is missing due openshift_facts.py breaking the dictionary as described in #6263 .

**Comment by djdevin** [comment]
I still get this without metrics, logging, or any mention of NFS. Hosted registry is using S3. Cleared all my facts and reran just to make sure, and I see

```
                "metrics": {
                    "deploy": "true",
                    "storage": {
                        "access": {
                            "modes": [
                                "ReadWriteOnce"
                            ]
                        },
                        "nfs": {
                            "directory": "/"
                        },
                        "volume": {
                            "name": "metrics",
                            "size": "1Gi"
                        }
                    }
                },
```
which is definitely wrong.

I have openshift_metrics_install_metrics=false, openshift_logging_install_logging=false

openshift-ansible 3.7 (latest as of today)

It could be an issue with a previous installation of metrics, this is an old cluster. I've run the uninstallation playbooks just to make sure, but it still happens.

**Comment by djdevin** [comment]
I was able to get this to work by manually editing `/etc/ansible/facts.d/openshift.fact` on the master and removing all sections of metrics/logging, then commenting out lines about metrics in the inventory.

Even though I had those *_install set to false, it still wrote things to facts.

My guess is that this persistent volume creation does not look at the *_install variables, but looks at the other ones (storage etc.). Once those get set I couldn't find a way to get them unset without manually editing facts.

**Comment by vrutkovs** [comment]
This was fixed in #7000. If this still occurs on latest released RPMs please open a new issue
